trieste/acquisition/interface.py

```diff
@@ -316,57 +304,24 @@ def update_acquisition_function(
         )
 
 
-class VectorizedAcquisitionFunctionBuilder(Generic[T], ABC):
+class VectorizedAcquisitionFunctionBuilder(AcquisitionFunctionBuilder[T]):
     """
     An :class:`VectorizedAcquisitionFunctionBuilder` builds and updates a vectorized
-    acquisition function.
+    acquisition function These differ from normal acquisition functions only by their output shape:
+    rather than returng a single value, they return one value per potential query point.
+    Thus, with leading dimensions, they take input shape `[..., B, D]` and returns shape `[..., B]`.
     """
 
-    @abstractmethod
-    def prepare_acquisition_function(
-        self,
-        models: Mapping[str, T],
-        datasets: Optional[Mapping[str, Dataset]] = None,
-    ) -> VectorizedAcquisitionFunction:
-        """
-        Prepare a vectorized acquisition function. We assume that this requires at least models, but
-        it may sometimes also need data.
-
-        :param models: The models for each tag.
-        :param datasets: The data from the observer (optional).
-        :return: An vectorized acquisition function.
-        """
-
-    def update_acquisition_function(
-        self,
-        function: VectorizedAcquisitionFunction,
-        models: Mapping[str, T],
-        datasets: Optional[Mapping[str, Dataset]] = None,
-    ) -> VectorizedAcquisitionFunction:
-        """
-        Update a vectorized acquisition function. By default this generates a new acquisition
-        function each time. However, if the function is decorated with `@tf.function`, then you
-        can override this method to update its variables instead and avoid retracing the acquisition
-        function on every optimization loop.
-
-        :param function: The vectorized acquisition function to update.
-        :param models: The models for each tag.
-        :param datasets: The data from the observer (optional).
-        :return: The updated vectorized acquisition function.
-        """
-        return self.prepare_acquisition_function(models, datasets=datasets)
 
-
-class SingleModelVectorizedAcquisitionBuilder(Generic[T], ABC):
+class SingleModelVectorizedAcquisitionBuilder(SingleModelAcquisitionBuilder[T]):
     """
-    Convenience acquisition function builder for a vectorized acquisition function (or component
+    Convenience acquisition function builder for vectorized acquisition functions (or component
     of a composite vectorized acquisition function) that requires only one model, dataset pair.
     """
 
-    def using(self, tag: str) -> VectorizedAcquisitionFunctionBuilder[T]:
+    def using(self, tag: str) -> AcquisitionFunctionBuilder[T]:
```

Comment:
I don't understand this change. Don't we want `SingleModelVectorizedAcquisitionBuilder().using(OBJECTIVE)` to return a `VectorizedAcquisitionFunctionBuilder` rather than a `AcquisitionFunctionBuilder` (so that we recognise it as such in the rule?)